### ISSUE TITLE
Add a "ignore" flag on objects to remove them from render

### ISSUE THREAD

**Issue by michelchafouin** [LuxCoreRender/BlendLuxCore#331]
This is a bit of a special case, but this would help being able to switch between Cycles/EEVEE and LuxCore.

Unfortunately, light shapes cast shadows and there is no way around it at the moment.
What I miss is a way to set LuxCore to ignore that light, so this light can be active for Cycles/EEVEE, but not in LuxCore.
I would suggest to have in the Object panel, next to Visible to Camera, a flag to set an object to be ignored and not exported for render.

Visibility flags in the outliner are not an option because they would affect Cycles/EEVEE as well.

**Comment by Dade916** [comment]
> Unfortunately, light shapes cast shadows and there is no way around it at the moment.

Objects can be light transparent (https://forums.luxcorerender.org/viewtopic.php?f=5&t=1474), is it what you are looking for ?

**Comment by michelchafouin** [comment]
But it is not available for lights, right?

**Comment by Dade916** [comment]
Do you want the light source to be visible but not to cast a shadow ? Or just to be invisible ?

**Comment by michelchafouin** [comment]
Completely invisible in that case.

**Comment by Dade916** [comment]
I have never tried but you can make an object camera invisible and shadow transparent (in that case it should be totally invisible).

**Comment by michelchafouin** [comment]
I know that this is a very specific case, as I'm trying to make an asset library of lamps that would be compatible with Cycles/EEVEE and LuxCore... The problem is that these lights are used in Cycles / EEVEE, I can't replace them by emissive objects, and I can't hide them or it will break Cycles / EEVEE renders.
Technically, if I could control light path visibility like in Cycles, I wouldn't have to hide them, but since I know this is a hard one with BiDir, I'm just trying to get them to be ignored by LuxCore so I can rely on emissive meshes instead.

**Comment by Theverat** [comment]
A generic "ignore" flag on object level could be added here: 
https://github.com/LuxCoreRender/BlendLuxCore/blob/cae55c85abdf67ffee66caf7baca604996e85812/properties/blender_object.py#L18
And the check during export would be placed here: 
https://github.com/LuxCoreRender/BlendLuxCore/blob/cae55c85abdf67ffee66caf7baca604996e85812/export/caches/object_cache.py#L86

**Comment by michelchafouin** [comment]
Thanks, I have added an "Exclude from render" flag locally. However, the viewport render isn't getting updated unless I restart it. How do we tell it to rebuild the scene?

**Comment by Theverat** [comment]
You probably have to add a check to the visibility cache: 
https://github.com/LuxCoreRender/BlendLuxCore/blob/cae55c85abdf67ffee66caf7baca604996e85812/export/caches/__init__.py#L85

Add your condition like this:
```diff
            if dg_obj_instance.show_self:
                obj = dg_obj_instance.instance_object if dg_obj_instance.is_instance else dg_obj_instance.object

-               if obj.type in EXPORTABLE_OBJECTS:
+               if not obj.luxcore.ignore and obj.type in EXPORTABLE_OBJECTS:
                    keys.add(utils.make_key_from_instance(dg_obj_instance))
```

**Comment by Theverat** [comment]
Implemented in a63892053c5794274066bf627f2c81e9b3a1bb42, e7950bba60df4ad9d92f58336a4e8b0cc8d76ef2, e273557e70391bc743e61a16cab0d42812203f60, ef455d04c33a30785123490e5e52fc7de6d99ed1 and 4997b9f825fd0e538385154df6db0fbe0e370fef.

Thanks for taking matters in your own hands!
(but next time just one pull request please ;)
